diff -ur gcc.vanilla/libstdc++-v3/configure gcc-3.3.3-200402017-1/libstdc++-v3/configure
--- gcc.vanilla/libstdc++-v3/configure	2004-01-21 21:03:26.000000000 +0100
+++ gcc-3.3.3-200402017-1/libstdc++-v3/configure	2004-07-02 16:55:51.000000000 +0200
@@ -5189,13 +5189,6 @@
 EOF
 
       ;;
-    *)
-      os_include_dir="config/os/newlib"
-      cat >> confdefs.h <<\EOF
-#define HAVE_HYPOT 1
-EOF
-
-      ;;
     *-linux*)
       os_include_dir="os/gnu-linux"
       for ac_hdr in nan.h ieeefp.h endian.h sys/isa_defs.h \

