diff -ru gnet-2.0.4-vanilla/src/makefile.mingw gnet-2.0.4/src/makefile.mingw
--- gnet-2.0.4-vanilla/src/makefile.mingw	2003-07-24 03:34:53.000000000 +0200
+++ gnet-2.0.4/src/makefile.mingw	2003-09-02 23:55:12.000000000 +0200
@@ -1,6 +1,5 @@
-CC = gcc
-FLAGS = -g -Wall -mno-cygwin -mcpu=pentium -DGNET_EXPERIMENTAL=1
-INCLUDE = -I./ `pkg-config --cflags glib-2.0`
+FLAGS = -O2 -g -Wall -mno-cygwin -mcpu=pentium -DGNET_EXPERIMENTAL=1
+INCLUDE = -I. -I.. `pkg-config --cflags glib-2.0`
 LIBS = `pkg-config --libs glib-2.0` -lws2_32
 OFILES = gnet-private.o gnet.o ipv6.o inetaddr.o iochannel.o tcp.o udp.o mcast.o socks-private.o socks.o conn.o server.o pack.o md5.o sha.o uri.o base64.o
 
@@ -22,4 +21,5 @@
 	$(CC) $(FLAGS) $(INCLUDE) -c sha.c
 	$(CC) $(FLAGS) $(INCLUDE) -c uri.c
 	$(CC) $(FLAGS) $(INCLUDE) -c base64.c
-	dllwrap $(INCLUDE) --export-all --output-def gnet.def --implib libgnet-2.0.a -o gnet-2.0.dll $(OFILES) $(LIBS)
+	${CC} ${FLAGS} ${INCLUDE} -shared -o gnet-2.0.dll -Wl,--export-all-symbols -Wl,--out-implib,libgnet-2.0.a  $(OFILES) $(LIBS)
+
