diff -ur gnet-2.0.7/src/gnet-private.c source/gnet-2.0.7/src/gnet-private.c
--- gnet-2.0.7/src/gnet-private.c	2005-02-19 18:15:51.000000000 +0100
+++ source/gnet-2.0.7/src/gnet-private.c	2006-01-28 15:39:21.000000000 +0100
@@ -275,14 +275,14 @@
 	    return FALSE;
 	  }
 
-	gnet_iochannel = g_io_channel_win32_new_messages((unsigned int)gnet_hWnd);
+/* 	gnet_iochannel = g_io_channel_win32_new_messages((unsigned int)gnet_hWnd); */
 
-	/* Add a watch */
+/* 	/\* Add a watch *\/ */
 	
-	gnet_io_watch_ID = g_io_add_watch(gnet_iochannel,
-					  (GIOCondition)(G_IO_IN|G_IO_ERR|G_IO_HUP|G_IO_NVAL),
-					  gnet_MainCallBack, 
-					  NULL);
+/* 	gnet_io_watch_ID = g_io_add_watch(gnet_iochannel, */
+/* 					  (GIOCondition)(G_IO_IN|G_IO_ERR|G_IO_HUP|G_IO_NVAL), */
+/* 					  gnet_MainCallBack,  */
+/* 					  NULL); */
 					  
 	break;
       }
@@ -301,8 +301,8 @@
     case DLL_PROCESS_DETACH:
       /* The DLL unmapped from process's address space. Do necessary cleanup */
       {
-	g_source_remove(gnet_io_watch_ID);
-	g_free(gnet_iochannel);
+/* 	g_source_remove(gnet_io_watch_ID); */
+/* 	g_free(gnet_iochannel); */
  	DestroyWindow(gnet_hWnd); 
 
 	/*CleanUp WinSock 2 */
diff -ur gnet-2.0.7/src/makefile.mingw source/gnet-2.0.7/src/makefile.mingw
--- gnet-2.0.7/src/makefile.mingw	2005-02-10 10:35:34.000000000 +0100
+++ source/gnet-2.0.7/src/makefile.mingw	2006-01-28 15:34:41.000000000 +0100
@@ -1,6 +1,6 @@
-CC = gcc
-FLAGS = -g -Wall -mno-cygwin -mcpu=pentium -DGNET_EXPERIMENTAL=1
-INCLUDE = -I./ `pkg-config --cflags glib-2.0`
+#CC = gcc
+FLAGS = -g -O2 -Wall -DGNET_EXPERIMENTAL=1
+INCLUDE = -I./ -I.. `pkg-config --cflags glib-2.0`
 LIBS = `pkg-config --libs glib-2.0` -lws2_32
 OFILES = gnet-private.o gnet.o ipv6.o inetaddr.o iochannel.o tcp.o udp.o mcast.o socks-private.o socks.o conn.o conn-http.o server.o pack.o md5.o sha.o uri.o base64.o
 
@@ -23,4 +23,9 @@
 	$(CC) $(FLAGS) $(INCLUDE) -c sha.c
 	$(CC) $(FLAGS) $(INCLUDE) -c uri.c
 	$(CC) $(FLAGS) $(INCLUDE) -c base64.c
-	dllwrap $(INCLUDE) --export-all --output-def gnet.def --implib libgnet-2.0.a -o gnet-2.0.dll $(OFILES) $(LIBS)
+	${CC} -shared -o gnet-2.0.dll -Wl,--out-implib,libgnet-2.0.a -Wl,--enable-auto-import -Wl,--whole-archive $(OFILES) -Wl,--no-whole-archive $(LIBS)
+	i386-mingw32msvc-dlltool --def ../gnet.def --dllname gnet-2.0.dll --output-lib libgnet-2.0.a
+
+##        i386-mingw32msvc-dllwrap $(INCLUDE) -def ../gnet.def --output-def gnet.def --implib libgnet-2.0.a -o gnet-2.0.dll $(OFILES) $(LIBS)
+
+
