diff --git a/dbus/dbus-sysdeps-win.c b/dbus/dbus-sysdeps-win.c
index 19f6fa0..91d5a03 100644
--- a/dbus/dbus-sysdeps-win.c
+++ b/dbus/dbus-sysdeps-win.c
@@ -2847,6 +2847,7 @@ _dbus_get_autolaunch_address (const char *scope, DBusString *address,
     {
       printf ("please add the path to %s to your PATH environment variable\n", daemon_name);
       printf ("or start the daemon manually\n\n");
+      dbus_set_error_const (error, DBUS_ERROR_FAILED, "Failed to locate dbus-daemon");
       goto out;
     }
 
@@ -3126,6 +3127,8 @@ _dbus_get_install_root(char *prefix, int len)
     //the Latin "bin", but that is not likely I think...
     if (lastSlash - prefix >= 4 && strnicmp(lastSlash - 4, "\\bin", 4) == 0)
         lastSlash[-3] = 0;
+    if (lastSlash - prefix >= 4 && strnicmp(lastSlash - 4, "\\lib", 4) == 0)
+        lastSlash[-3] = 0;
     else if (lastSlash - prefix >= 10 && strnicmp(lastSlash - 10, "\\bin\\debug", 10) == 0)
         lastSlash[-9] = 0;
     else if (lastSlash - prefix >= 12 && strnicmp(lastSlash - 12, "\\bin\\release", 12) == 0)
